### DISCUSSION TITLE
Hermes and alternative JS engines

### DISCUSSION THREAD

**Discussion by vjpr** [oven-sh/bun#809] [General]
I wonder if Bun considered using [Hermes](https://hermesengine.dev/) as the JS engine (or whether it would allow supporting multiple engines).

Hermes is probably the newest engine to be written, and focuses on fast startup time and lean memory usage. 

This could be important for the cloud edge function use case - as I've heard is a good use case for Bun.

Also makes it great for WebAssembly targets and running in the browser.

NOTE: Hermes outside of RN is discussed here: https://github.com/facebook/hermes/issues/326

**Comment by siemato** [comment]
I'd assume, Hermes just doesn't fit the use case. AFAIK, Hermes is built around the assumption of generating some sort of compiled static representation of a subset of JS as a build step. Bun IS the build time though. According to this premise, Hermes should be used as an efficient runtime for already packaged code only. Open to criticism if my assumptions are wrong :)

**Reply by Bellisario** [reply]
Maybe if more efficient but only good for bundled code, could be used into a Bun production version, as @vjpr was asking on #812:
> Does bun have a production-only build that could be even slimmer?

**Reply by siemato** [reply]
After thinking about it, it's less and less an option. As far as I understand, there's no wasm support, no webgpu, no nothing. it's integrating well into React Native, but seems of very limited use outside of RN for now. I can see benefits of precompiled JS for edge compute/functions, so I'm not dismissing the idea itself. Startup times are critical in that area of application.

**Comment by filipef101** [comment]
@siemato With static hermes announcement, and corresponding performance claims, could it be worth re-considering hermes? 
What is an use case for WASM and webgpu with bun runtime? Would be nice to either allow both runtimes, then allowing to hook into engine specific API's & WASM.

**Reply by siemato** [reply]
it never was about hermes itself being a bad choice, my comment is more about the ecosystem surrounding it, and the effort required. With JavaScriptCore, it's possible to piggy-back on Apple's effort of creating native platform apis. AFAIK, Hermes piggy-backs on Android NDK, to implement HTTP/File Access ec (but I might be wrong here, it's been a year or so, since I looked into hermes, and my memory's getting faint)
